pkg/util/helpers.go

```diff
@@ -36,3 +39,55 @@ type Container struct {
 	LivenessProbeConfigs  []*horizonapi.ProbeConfig
 	PreStopConfig         *horizonapi.ActionConfig
 }
+
+// MergeEnvMaps will merge the source and destination environs. If the same value exist in both, destination environ will given more preference
+func MergeEnvMaps(source, destination map[string]string) map[string]string {
+	// if the source key present in the destination map, it will overrides the destination value
+	// if the source value is empty, then delete it from the destination
+	for key, value := range source {
+		if len(value) == 0 {
+			delete(destination, key)
+		} else {
+			destination[key] = value
+		}
+	}
+	return destination
+}
+
+// MergeEnvSlices will merge the source and destination environs. If the same value exist in both, destination environ will given more preference
```

Comment:
It seems like the source environ is given preference. If a source value is provided then it will override the destination value

Reply:
fixed